src/fcp/specs/v2.py

```diff
@@ -100,6 +100,10 @@ def get_matching_extension(self, struct: Struct, protocol: str) -> Maybe[Impl]:
 
         return Nothing()
 
+    def get_matching_extensions(self, protocol: str) -> List[Impl]:
+        """Get impl for corresponding struct with a specific protocol."""
+        return [extension for extension in self.impls]
+
```

Comment:
`get_matching_impls`

I've renamed `extension` to  `impl` following the syntax change. Anyway the protocol argument is not being used currently